### DISCUSSION TITLE
Suggestions for Minimizing (or Eliminating) Risk of OKD on Several E8450 Routers

### DISCUSSION THREAD

**Discussion by foxrid3r** [dangowrt/owrt-ubi-installer#195] [Q&A] [answered]
Looking for some guidance on how to reduce the risk of OKD on several E8450 routers I have which have all been flashed with OpenWRT. I have 8 of these routers which I've flashed using the v1.0.3 installer and show TF-A v2.9 firmware.

![image](https://github.com/user-attachments/assets/ef2801a4-96c3-4135-833d-87b972de657a)

These routers are being used as wireless access points to wirelessly access machines with local networks. All routers have vsftpd FTP servers running on them and will hand out DHCP addresses to wireless clients when they connect. There isn't anything fancy being done with these device; we just use them to easily connect to different machines at work and to provide local FTP servers for devices to write files to.

I've had one router already experience the OKD and another one that I suspect might fall victim to it soon. I'm looking for suggestions on what I can do to minimize the risk of these devices failing in the future. I'm looking to make these devices as reliable and stable as possible.

I am somewhat of a beginner when it comes to OpenWRT, so detailed suggestions or links would be greatly appreciated!

**Comment by dangowrt** [comment, marked as answer]
You should downgrade `bl2` to v2.4. I've added a section to the Wiki explaining how to do this:
https://openwrt.org/toh/linksys/e8450#recovery_from_live_openwrt_2203x_or_2305x_with_fip_in_mtd_layout

Note that upcoming 24.10.0 release will make things much more robust, however, in order to update to that you will need to run on updated installer. Up to the point of the 24.10.0 release I recommend to just downgrade bl2 to v2.4 as described above.
